File: starters/javascript/MyTronBot.js
// Template for your tron bot

var tron = require('./tron');

// Return one element of the array a at random

function randomChoice(a) {
    return a[Math.floor(Math.random() * a.length)];
}

function which_move(board) {

    // fill in your code here. it must return one of the following directions:
    //   tron.NORTH, tron.EAST, tron.SOUTH, tron.WEST

    // For now, choose a legal move randomly.
    // Note that board.moves will produce [NORTH] if there are no
    // legal moves available.

    return randomChoice(board.moves());
}

// you do not need to modify this part
tron.play(which_move);

File: starters/go/MyTronBot.go
// MyTronBot.go
//
// This is the code file that you will modify in order to create your entry.

package main

func MakeMove(board *Board) int {
	x, y := board.MyXY()
	if board.IsPassable(x, y-1) {
		return NORTH
	}
	if board.IsPassable(x+1, y) {
		return EAST
	}
	if board.IsPassable(x, y+1) {
		return SOUTH
	}
	if board.IsPassable(x-1, y) {
		return WEST
	}
	return NORTH
}

// Ignore this function. It is just handling boring stuff for you, like
// communicating with the Tron tournament engine.
func main() {
	for {
		SendMove(MakeMove(NewBoard()))
	}
}
